From 7af0102388fe9ede6431d21df483dc7bf5c47d0f Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@homa.ne.jp>
Date: Mon, 31 Oct 2016 23:10:47 +0900
Subject: [PATCH 3/3] PDMFactory: Add a fallback blank decoder module

It will be used only when there is no appropriate decoder while the
normal blank decoder module always be used when it's enabled.

Signed-off-by: Takuro Ashie <ashie@homa.ne.jp>
---
 dom/media/platforms/PDMFactory.cpp | 8 ++++++++
 dom/media/platforms/PDMFactory.h   | 1 +
 2 files changed, 9 insertions(+)

diff --git a/dom/media/platforms/PDMFactory.cpp b/dom/media/platforms/PDMFactory.cpp
index 1750e50..2a78253 100644
--- a/dom/media/platforms/PDMFactory.cpp
+++ b/dom/media/platforms/PDMFactory.cpp
@@ -48,6 +48,7 @@ extern already_AddRefed<PlatformDecoderModule> CreateAgnosticDecoderModule();
 extern already_AddRefed<PlatformDecoderModule> CreateBlankDecoderModule();
 
 bool PDMFactory::sUseBlankDecoder = false;
+bool PDMFactory::sUseFallbackBlankDecoder = false;
 #ifdef MOZ_GONK_MEDIACODEC
 bool PDMFactory::sGonkDecoderEnabled = false;
 #endif
@@ -83,6 +84,8 @@ PDMFactory::Init()
 
   Preferences::AddBoolVarCache(&sUseBlankDecoder,
                                "media.use-blank-decoder");
+  Preferences::AddBoolVarCache(&sUseFallbackBlankDecoder,
+                               "media.use-fallback-blank-decoder");
 #ifdef MOZ_GONK_MEDIACODEC
   Preferences::AddBoolVarCache(&sGonkDecoderEnabled,
                                "media.gonk.enabled", false);
@@ -310,6 +313,11 @@ PDMFactory::CreatePDMs()
     m = new GMPDecoderModule();
     StartupPDM(m);
   }  
+
+  if (sUseFallbackBlankDecoder) {
+    m = CreateBlankDecoderModule();
+    StartupPDM(m);
+  }
 }
 
 bool
diff --git a/dom/media/platforms/PDMFactory.h b/dom/media/platforms/PDMFactory.h
index 3c51922..3f7bdff2c 100644
--- a/dom/media/platforms/PDMFactory.h
+++ b/dom/media/platforms/PDMFactory.h
@@ -65,6 +65,7 @@ private:
 
   // PDM pref caches...
   static bool sUseBlankDecoder;
+  static bool sUseFallbackBlankDecoder;
 #ifdef MOZ_GONK_MEDIACODEC
   static bool sGonkDecoderEnabled;
 #endif
-- 
1.9.1

